elide-contrib/elide-swagger/src/main/java/com/yahoo/elide/contrib/swagger/JsonApiModelResolver.java

```diff
@@ -0,0 +1,172 @@
+import io.swagger.models.Model;
+import io.swagger.models.properties.Property;
+import io.swagger.util.Json;
+
+import java.lang.reflect.Type;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Swagger ModelResolvers map POJO classes to Swagger com.yahoo.elide.contrib.swagger.models.
+ * This resolver maps the POJO to a JSON-API Resource.
+ */
+public class JsonApiModelResolver extends ModelResolver {
+    EntityDictionary dictionary;
+
+    public JsonApiModelResolver(EntityDictionary dictionary) {
+        super(Json.mapper());
+        this.dictionary = dictionary;
+    }
+
+    @Override
+    public Model resolve(Type type, ModelConverterContext context, Iterator<ModelConverter> next) {
+        if (!(type instanceof Class)) {
+            return super.resolve(type, context, next);
+        }
+
+        Class<?> clazz = (Class<?>) type;
+        String typeAlias = dictionary.getJsonAliasFor(clazz);
+
+        /* Not an entity managed by Elide, let Swagger convert it */
+        if (typeAlias == null) {
+            return super.resolve(type, context, next);
+        }
+
+        Resource entitySchema = new Resource();
+        entitySchema.setSecurityDescription(getClassPermissions(clazz));
+
+        /* Populate the attributes */
+        List<String> attributeNames = dictionary.getAttributes(clazz);
+        for (String attributeName : attributeNames) {
+            Class<?> attributeClazz = dictionary.getType(clazz, attributeName);
+
+            Property attribute = super.resolveProperty(attributeClazz, context, null, next);
+
+            attribute.setDescription(getFieldPermissions(clazz, attributeName));
+            entitySchema.addAttribute(attributeName, attribute);
+        }
+
+        /* Populate the relationships */
+        List<String> relationshipNames = dictionary.getRelationships(clazz);
+        for (String relationshipName : relationshipNames) {
+
+            Class<?> relationshipType = dictionary.getParameterizedType(clazz, relationshipName);
+            Relationship relationship = new Relationship(dictionary.getJsonAliasFor(relationshipType));
+            relationship.setDescription(getFieldPermissions(clazz, relationshipName));
+
+            entitySchema.addRelationship(relationshipName, relationship);
+        }
+
+        entitySchema.name(typeAlias);
+        return entitySchema;
+    }
+
+    /**
+     * @param clazz the entity class
+     * @return the create and delete permissions for the entity class.
+     */
+    protected String getClassPermissions(Class<?> clazz) {
+        String createPermissions = getCreatePermission(clazz);
+        String deletePermissions = getDeletePermission(clazz);
+
+
+        createPermissions = (createPermissions == null) ? "" : "Create Permissions : (" + createPermissions + ")\n";
+        deletePermissions = (deletePermissions == null) ? "" : "Delete Permissions : (" + deletePermissions + ")\n";
+        return createPermissions + deletePermissions;
```

Comment:
Should we add an expanded `ReadPermission` here? One that shows the `OR` of the _unique_ clauses (i.e. ordering-agnostic whenever possible) among the `class`, `attribute`, and `relationship` permissions? 

This may ultimately be useful for understanding under what set of conditions _something_ will be returned from the model. That is, if I start seeing 404's, I can see at-a-glance whether I should have access under _any_ possible condition. Field permissions can then be drilled down more directly for further debugging.